### PR TITLE
draw minus

### PR DESCRIPTION
**Pull request by philippTheCat**

### CODE REVIEW

File: BunnymodXT/hud_custom.cpp [hunk 1]
```diff
@@ -113,16 +113,137 @@ namespace CustomHud
...
+			89
+		};
+
+		const int Line640_Right[] = {
+			 83,  21,
+			 255, 104,
+			 255, 192,
+			 255, 165,
+			 255, 65,
+			  29,  29
```
Comment (YaLTeR) [inline BunnymodXT/hud_custom.cpp]: Extra spaces before each line here

File: BunnymodXT/hud_custom.cpp [hunk 2]
```diff
@@ -113,16 +113,137 @@ namespace CustomHud
...
+		if (si.iWidth < 640)
+			DrawBitmap(x, y, Dot320, 3, 3, r, g, b);
+		else
+			DrawBitmap(x, y, Dot640, 6, 6, r, g, b);
+	}
+
+	static void DrawLine(int x, int y, int length, int r, int g, int b) {
+		if ((si.iWidth < 640 && length < 3) || length < 5) {
+			// might as well draw a dot
+			DrawDot(x, y, r, g, b);
```
Comment (YaLTeR) [inline BunnymodXT/hud_custom.cpp]: Forgot return?

File: BunnymodXT/hud_custom.cpp [hunk 3]
```diff
@@ -113,16 +113,137 @@ namespace CustomHud
...
+		const int Line320_left[] = {
+			143,
+			255,
+			120,
+		};
+
+		const int Line320_repeat[] = {
+			 199,
+			 255,
+			 169,
```
Comment (YaLTeR) [inline BunnymodXT/hud_custom.cpp]: Extra spaces

File: BunnymodXT/hud_custom.cpp [hunk 4]
```diff
@@ -113,16 +113,137 @@ namespace CustomHud
...
+		};
+
+		const int Line320_Right[] = {
+			 122,
+			 218,
+			 95
```
Comment (YaLTeR) [inline BunnymodXT/hud_custom.cpp]: Extra spaces

File: BunnymodXT/hud_custom.cpp [hunk 5]
```diff
@@ -113,16 +113,137 @@ namespace CustomHud
...
+		int linesprite_width;
+
+
+		if (si.iWidth <= 640) {
+			Line_Left = Line320_left;
+			Line_Repeat = Line320_repeat;
+			Line_Right = Line320_Right;
+			linesprite_height = 3;
+			linesprite_width = 1;
+		}
```
Comment (YaLTeR) [inline BunnymodXT/hud_custom.cpp]: else should be on the same line as {

File: BunnymodXT/hud_custom.cpp [hunk 6]
```diff
@@ -113,16 +113,137 @@ namespace CustomHud
...
+			Line_Right = Line640_Right;
+			linesprite_height = 6;
+			linesprite_width = 2;
+		}
+
+		y -= linesprite_height / 2;
+
+		x += DrawBitmap(x, y, Line_Left, linesprite_width, linesprite_height, r, g, b);
+
+
```
Comment (YaLTeR) [inline BunnymodXT/hud_custom.cpp]: Extra blank line

File: BunnymodXT/hud_custom.cpp [hunk 7]
```diff
@@ -113,16 +113,137 @@ namespace CustomHud
...
+		const int Line320_Right[] = {
+			 122,
+			 218,
+			 95
+		};
+
+		const int *Line_Left;
+		const int *Line_Repeat;
+		const int *Line_Right;
```
Comment (YaLTeR) [inline BunnymodXT/hud_custom.cpp]: Don't use uppercase letters for these vars

File: BunnymodXT/hud_custom.cpp [hunk 8]
```diff
@@ -113,16 +113,137 @@ namespace CustomHud
...
+
+		x += DrawBitmap(x, y, Line_Left, linesprite_width, linesprite_height, r, g, b);
+
+
+		for (int xOffset = 0; xOffset < length - linesprite_width * 2; xOffset++) {
+			x += DrawBitmap(x, y, Line_Repeat, 1, linesprite_height, r, g, b);
+		}
+
+		DrawBitmap(x, y, Line_Right, linesprite_width, linesprite_height, r, g, b);
+
```
Comment (YaLTeR) [inline BunnymodXT/hud_custom.cpp]: Extra blank line

File: BunnymodXT/hud_custom.cpp [hunk 9]
```diff
@@ -113,16 +113,137 @@ namespace CustomHud
...
+			 95
+		};
+
+		const int *Line_Left;
+		const int *Line_Repeat;
+		const int *Line_Right;
+		int linesprite_height;
+		int linesprite_width;
+
+
```
Comment (Matherunner) [inline BunnymodXT/hud_custom.cpp]: Also delete this extra blank line

**Comment by philippTheCat** [comment]
Regarding performance: I can do about 500 Dots (1 DrawBitmap call with a 6x6 bitmap -> 36 draw calls _each_) ->18 000Drawcalls at about 70fps. 

The drawMinus method uses 72 calls for resolutions smaller than 640 (12* 6), and 144 for bigger resolutions (24*6)

**Comment by Matherunner** [comment]
@philippTheCat A more useful metric is how adding the minus sign impacts the **frame rate**, rather than how many draws it can make per second.

**Comment by philippTheCat** [comment]
It doesnt have any measurable performance impact. I get a stable 1000fps, without and with the minus sign.

The DrawNumber method is also only used in 4 places: Jumpspeed, Health, Speedometer and the timer. Only the health one can be minus, and I would argue that performance isnt that important when you're dead

**Pull request merged**